### PR TITLE
RavenDB-5531 V3.0 back compatibility to version v4.0

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs [hunk 1]
```diff
@@ -1012,6 +1012,12 @@ public static void WriteLastEtagsToFile(OperationState result, string backupPath
                 return deletedDocumentsCount;
             });
 
+            exportSectionRegistar.Add(Constants.BuildVersion, async () =>
+            {
```
Comment (ayende) [inline Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1016]: Need to add a warning here.

File: Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs [hunk 2]
```diff
@@ -1061,7 +1067,8 @@ public static void WriteLastEtagsToFile(OperationState result, string backupPath
                     continue;
                 }
 
-                if (jsonReader.TokenType != JsonToken.StartArray)
+                if (jsonReader.TokenType != JsonToken.StartArray &&
+                    !currentSection.Equals(Constants.BuildVersion))
```
Comment (ayende) [inline Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1072]: You shouldn't have to do that, consume the value in the BuildVersion section

**Review by ayende — CHANGES_REQUESTED** [review]

File: Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs [hunk 3]
```diff
@@ -1014,6 +1014,7 @@ public static void WriteLastEtagsToFile(OperationState result, string backupPath
 
             exportSectionRegistar.Add(Constants.BuildVersion, async () =>
             {
+                Operations.ShowProgress("Skipping import of indexes");
                 Options.OperateOnTypes &= ~(ItemType.Indexes | ItemType.Transformers);
```
Comment (ayende) [inline Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs]: You are also skipping transformers

**Review by ayende — CHANGES_REQUESTED** [review]

File: Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs [hunk 4]
```diff
@@ -1014,6 +1014,7 @@ public static void WriteLastEtagsToFile(OperationState result, string backupPath
 
             exportSectionRegistar.Add(Constants.BuildVersion, async () =>
             {
+                Operations.ShowProgress("Skipping import of indexes");
```
Comment (ppekrol) [inline Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1017]: You need to put exact reason here. Maybe "Detected 4.x file format. Automatically disabled index and transformer import." ?

File: Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs [hunk 5]
```diff
@@ -1259,7 +1260,7 @@ private Task<int> ImportIdentities(JsonTextReader jsonReader)
                 if ((Options.OperateOnTypes & ItemType.Transformers) != ItemType.Transformers)
                     continue;
 
-                var transformerName = transformer.Value<string>("name");
+                var transformerName = transformer.Value<string>("name") ?? transformer.Value<string>("Name");
```
Comment (ppekrol) [inline Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1263]: Why you added this?

Comment (ayende) [reply Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1263]: So we'll have nicer syntax for 4.0

Comment (ppekrol) [reply Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1263]: But we are not importing 4.0 transformers?

Comment (IdanHaim) [reply Raven.Abstractions/Smuggler/SmugglerDatabaseApiBase.cs:1263]: We are now

**Comment by ayende** [comment]
Manually merged

**Pull request closed without merging**